hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/client/RMProxy.java

```diff
@@ -154,19 +151,44 @@ public T run() {
       });
   }
 
+
+  /**
+   * Helper method to create non-HA RMFailoverProxyProvider.
+   */
+  private <T> RMFailoverProxyProvider<T> createNonHaRMFailoverProxyProvider(
+      Configuration conf, Class<T> protocol) {
+    String defaultProviderClassName =
+        YarnConfiguration.DEFAULT_CLIENT_FAILOVER_NO_HA_PROXY_PROVIDER;
+    Class<? extends RMFailoverProxyProvider<T>> defaultProviderClass;
+    try {
+      defaultProviderClass = (Class<? extends RMFailoverProxyProvider<T>>)
+          Class.forName(defaultProviderClassName);
+    } catch (Exception e) {
+      throw new YarnRuntimeException("Invalid default failover provider class" +
+          defaultProviderClassName, e);
+    }
+
+    RMFailoverProxyProvider<T> provider = ReflectionUtils.newInstance(
+        conf.getClass(YarnConfiguration.CLIENT_FAILOVER_NO_HA_PROXY_PROVIDER,
+            defaultProviderClass, RMFailoverProxyProvider.class), conf);
+    provider.init(conf, (RMProxy<T>) this, protocol);
+    return provider;
+  }
+
   /**
    * Helper method to create FailoverProxyProvider.
    */
   private <T> RMFailoverProxyProvider<T> createRMFailoverProxyProvider(
       Configuration conf, Class<T> protocol) {
+    String defaultProviderClassName =
```

Comment:
Not sure is worth doing this change.

Reply:
We need this change so that the AutoRefreshNoHARMFailoverProxyProvider works. Otherwise, it will try to get the IP address once and then fail if there is ever a disconnect from the RM later

Reply:
I don't get it, aren't you just extracting the constant here and having the same behavior as before?

Reply:
My bad I thought this comment referred to a different issue.
I will change it back